flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -467,6 +487,128 @@ public void registerAtResourceManager(final String address) {
 		//TODO:: register at the RM
 	}
 
+	@RpcMethod
+	public NextInputSplit requestNextInputSplit(final JobVertexID vertexID, final ExecutionAttemptID executionAttempt) {
+		final byte[] serializedInputSplit;
+
+		final Execution execution = executionGraph.getRegisteredExecutions().get(executionAttempt);
+		if (execution == null) {
+			log.error("Can not find Execution for attempt {}.", executionAttempt);
```

Comment:
The log could be on "debug" level, as this situation can occur when the JobManager failed tasks and unregistered them, but the cancel() messages have not reached the TaskManager.

Since this is a valid situation, it should not cause a "WARN" message. In general, all situations that are valid race conditions and that need not indicate a corrupted state or lead to a failure / recovery should probably not have a "WARN" level.